### PR TITLE
refactor: move cli to plugin for consistency in arguments

### PR DESCRIPTION
**Pull request by jerrevanveluw**

### CODE REVIEW

File: src/plugin/arguments/src/commonMain/kotlin/community/flock/wirespec/plugin/CompilerArguments.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package community.flock.wirespec.plugin
+
+data class CompilerArguments(
```
Comment (ZzAve) [inline src/plugin/arguments/src/commonMain/kotlin/community/flock/wirespec/plugin/CompilerArguments.kt:3]: This is not about compiler arguments anymore, is it? But rather about plugin arguments?

File: src/plugin/arguments/build.gradle.kts
Comment (ZzAve) [inline src/plugin/arguments/build.gradle.kts:1]: What would be the succinct description of what the `arguments` module is about?

File: src/plugin/cli/src/commonMain/kotlin/community/flock/wirespec/plugin/cli/io/Extension.kt
Comment (ZzAve) [inline src/plugin/cli/src/commonMain/kotlin/community/flock/wirespec/plugin/cli/io/Extension.kt:1]: I believe we can safely drop this one? A rename to `FileExtension` happened. Not sure if intension was to put that in a separate file?

File: src/plugin/arguments/build.gradle.kts
Comment (jerrevanveluw) [reply src/plugin/arguments/build.gradle.kts:1]: could also be named 'common' or 'shared'

File: src/plugin/arguments/src/commonMain/kotlin/community/flock/wirespec/plugin/Input.kt [hunk 2]
```diff
@@ -11,13 +10,12 @@ sealed interface Input
 
 data class FullDirPath(val path: String) : Input
 
-@OptIn(ExperimentalStdlibApi::class)
```
[the remark this answers is not in the document]
Comment (jerrevanveluw) [reply src/plugin/arguments/src/commonMain/kotlin/community/flock/wirespec/plugin/Input.kt:14]: IDE didn't want it anymore

File: src/plugin/cli/src/commonMain/kotlin/community/flock/wirespec/plugin/cli/io/Extension.kt
Comment (jerrevanveluw) [reply src/plugin/cli/src/commonMain/kotlin/community/flock/wirespec/plugin/cli/io/Extension.kt:1]: Yes could be dropped

**Pull request merged**